pandas/core/arrays/interval.py

```diff
@@ -695,8 +696,10 @@ def astype(self, dtype, copy=True):
 
             # need to cast to different subtype
             try:
-                new_left = self.left.astype(dtype.subtype)
-                new_right = self.right.astype(dtype.subtype)
+                # We need to use Index rules for astype to prevent casting
+                #  np.nan entries to int subtypes
+                new_left = Index(self.left).astype(dtype.subtype)
```

Comment:
why don't we use a pair of Int array always? rather than a 2D
to avoid casting

Reply:
im planning to try that eventually, this is just trying to get everything working (still have a few failing view/copy tests)

Reply:
can we not astype directly on the array?

Reply:
doing it directly failed, need astype_nansafe.  havent checked if we can get way with just using that. (also found that PandasArray.astype needs to use astype_nansafe)